sdk/identity/Azure.Identity/src/ManagedIdentitySource.cs

```diff
@@ -67,6 +67,15 @@ public virtual async ValueTask<AccessToken> AuthenticateAsync(bool async, TokenR
                 exception = e;
             }
 
+            if (response.Status == 403)
+            {
+                string message = response.Content.ToString();
+                if (message.Contains("A socket operation was attempted to an unreachable network"))
```

Comment:
I'd add a comment for future maintainers explaining that this is for the Docker Container edge case